orangecontrib/educational/widgets/owgradientdescent.py

```diff
@@ -297,6 +350,51 @@ def __init__(self):
         self.step_size_lock()
         self.step_back_button_lock()
 
+    def _opt_changed(self):
+        self._show_right_optimizer()
+        self.select_optimizer()
+
+    def _show_right_optimizer(self):
+        enabled = [[False, False, False, False],  # SGD
+                   [True, False, False, False],  # Momentum
+                   [True, False, False, False],  # NAG
+                   [False, False, False, False],  # AdaGrad
+                   [False, True, False, False],  # RMSprop
+                   [False, True, False, False],  # AdaDelta
+                   [False, False, True, True],  # Adam
+                   [False, False, True, True],  # Adamax
+                ]
+
+        mask = [False, False, False, False]
+        self.combobox_opt.box.setVisible(self.stochastic)
+        if self.stochastic:
+            mask = enabled[self.opt_type]
+
+        # All hidden
+        for spin, enabled in zip(self._opt_params, mask):
+            [spin.box.hide, spin.box.show][enabled]()
+
+    def select_optimizer(self):
+        if self.learner is not None:
+            if self.opt_type == self._Optimizer.SGD:
+                self.learner.sgd_optimizer = Momentum(self.momentum)
+            elif self.opt_type == self._Optimizer.NAG:
+                self.learner.sgd_optimizer = NesterovMomentum(self.momentum)
+            elif self.opt_type == self._Optimizer.ADAGRAD:
+                self.learner.sgd_optimizer = AdaGrad()
+            elif self.opt_type == self._Optimizer.RMSPROP:
+                self.learner.sgd_optimizer = RMSProp(self.rho)
+            elif self.opt_type == self._Optimizer.ADADELTA:
+                self.learner.sgd_optimizer = AdaDelta(self.rho)
+            elif self.opt_type == self._Optimizer.ADAM:
+                self.learner.sgd_optimizer = Adam(beta1=self.beta1,
+                                                  beta2=self.beta2)
+            elif self.opt_type == self._Optimizer.ADAMAX:
+                self.learner.sgd_optimizer = Adamax(beta1=self.beta1,
+                                                    beta2=self.beta2)
+            else:
+                self.learner.sgd_optimizer = SGD()
```

Comment:
Usually, the learning rate is set when the optimizer is instantiated. But here, it is set to its corresponding value in each iteration.

Let's presume this case:
1. Select stochastic -> Vanilla SGD selected by default, with learning rate == 1.0
2. Step/Run -> Learning rate = self.learner.alpha (In this way, a user can change the learning rate at every moment, as well as the other parameters with no need to restart the process. This allow you to take an scheduled approach, e.g anneling.)